pkg/controllers/state/cluster.go

```diff
@@ -82,6 +82,16 @@ func NewCluster(clk clock.Clock, client client.Client) *Cluster {
 //
 //nolint:gocyclo
 func (c *Cluster) Synced(ctx context.Context) (synced bool) {
+	start := time.Now()
+	ClusterStateUnsyncedTimeSeconds.With(map[string]string{}).Set(time.Since(start).Seconds())
+	// Set/Delete the metric depending on the result of the Synced() call
+	defer func() {
+		if synced {
+			ClusterStateUnsyncedTimeSeconds.Delete(map[string]string{})
```

Comment:
I don't think we should delete this metric when we are synced -- we should just set the metric to 0